### DISCUSSION TITLE
Volumes mounted as read-only, is this intentional?

### DISCUSSION THREAD

**Discussion by nathanielks** [lima-vm/lima#393] [Q&A] [answered]
I was curious to see what sort of filesystem performance gains there were compared to Docker for Mac. I wanted to test a repo that would make a lot of filesystem writes, but in my tests I encountered an error saying the volume was mounted read-only. My apologies for my ignorance, but is this intentional? Should containers launched via lima with volumes be mounted as read-only volumes? For reference, here's how to reproduce:

```
cd path/to/somewhere
git clone git@github.com:withspectrum/spectrum.git
cd spectrum
lima nerdctl run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node:14 node shared/install-dependencies.js
```

This will eventually print the following error:

```
error Could not write file "/usr/src/app/yarn-error.log": "EROFS: read-only file system, open '/usr/src/app/yarn-error.log'"
error An unexpected error occurred: "EROFS: read-only file system, mkdir '/usr/src/app/node_modules'".
```

The docker analog that works ok:
```
docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node:14 node shared/install-dependencies.js
```

**Comment by AkihiroSuda** [comment]
https://github.com/AkihiroSuda/lima/blob/41fd9cc6a1e2bac73666e1f2b11604c7c42227dc/pkg/limayaml/default.TEMPLATE.yaml#L33-L41

The home is read-only by default.

**Comment by nathanielks** [comment]
Aha, I missed that! Thank you 🙇

**Comment by msolimans** [comment, marked as answer]
by default it is read only. to be make it writeable by default do the following: 

- edit the file and set `write: true` under `mount` section  

`$ vim  ~/.lima/default/lima.yaml` 

- then restart lima 

```
limactl list #this lists all running vms 
limactl stop default #or name of the machine 
limactl start default #or name of the machine 
```

**Reply by anschnapp** [reply]
Ok that would work but it feels like i would ignore the warnings in the default configuration file if i do so:

```
    # CAUTION: `writable` SHOULD be false for the home directory.
    # Setting `writable` to true is possible, but untested and dangerous.
```

Is there a reason why it's not supported in a tested fashion?

Is there a workaround to get a nerdctl volume bind running with write mode from within a sub directory of home without modifying the image definition (for maybe only adding this subdir as writeable) and rebuild the hole system?

**Reply by afbjorklund** [reply]
Doing a mount within a mount is left TODO; see #302

**Reply by anschnapp** [reply]
Great thanks for the information!

**Comment by afbjorklund** [comment]
> I was curious to see what sort of filesystem performance gains there were compared to Docker for Mac.

Pretty sure that the current remote file system (sshfs) is slower than the one used in docker (fuse + grpc) ?

If you want _performance_, you probably need to use a filesystem local to the VM - as discussed in #20

**Reply by afbjorklund** [reply]
That is, run the whole thing in a `lima` shell. And not just the nerdctl ?
(You might have to install `git` first, using `apt`)
